instrumentation/rocketmq-client-4.8/library/src/main/java/io/opentelemetry/instrumentation/rocketmq/TracingMessageInterceptor.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.rocketmq;
+
+import io.opentelemetry.api.trace.Span;
+import java.util.List;
+import org.apache.rocketmq.client.consumer.listener.ConsumeConcurrentlyStatus;
+import org.apache.rocketmq.client.consumer.listener.ConsumeOrderlyStatus;
+import org.apache.rocketmq.common.message.Message;
+import org.apache.rocketmq.common.message.MessageExt;
+
+public class TracingMessageInterceptor {
```

Comment:
Hmm, this class does not seem to be that useful to anybody using the library. Do we even need a library instrumentation if the particular lib (rocketmq) cannot be easily instrumented? @anuraaga @iNikem @trask

Reply:
Rocketmq seems to have some functionality to support tracing.

https://www.github.com/apache/rocketmq/tree/master/client%2Fsrc%2Fmain%2Fjava%2Forg%2Fapache%2Frocketmq%2Fclient%2Ftrace

https://github.com/apache/rocketmq/wiki/RIP-6-Message-Trace

If we can use it to provide library instrumentation and have the agent code just register it that's awesome. Agree that if a library prevents no way to hook in for instrumentation agent only is fine (unavoidable).

Reply:
Nice, that looks promising - using `ConsumeMessageHook` & `SendMessageHook` for otel tracing might be a good idea.

Reply:
@addname Would you be able to look into using hooks to instrument in library instrumentation, and then just registering that in the javaagent instrumentation? I would probably start with a PR that only includes library instrumentation if this looks feasible.

Reply:
@anuraaga OK, but registering hooks in the javaagent instrumentation seems a little difficult.

Reply:
@anuraaga @mateuszrzeszutek That's cool! I've tried and it is available use hooks to register in the iavaagent instrumentation. Later, I will reconstruct javaagent, library, and testing, totally 3 modules, and then commit them.